src/careamics/lvae_training/dataset/multifile_dataset.py

```diff
@@ -0,0 +1,274 @@
+            test_fraction=test_fraction,
+        )
+
+    @property
+    def data_path(self):
+        return self._fpath
+
+    def rm_bkground_set_max_val_and_upperclip_data(self, max_val, datasplit_type):
+        pass
+
+    def load_data(
+        self,
+        data_config: Union[MultiChDatasetConfig, LCDatasetConfig],
+        datasplit_type: DataSplitType,
+        load_data_fn: Callable[..., NDArray],
+        val_fraction=None,
+        test_fraction=None,
+        allow_generation=None,
+    ):
+        self._data = self._preloaded_data
+        assert "channel_1" not in data_config or isinstance(data_config.channel_1, str)
+        assert "channel_2" not in data_config or isinstance(data_config.channel_2, str)
+        assert "channel_3" not in data_config or isinstance(data_config.channel_3, str)
+        self._loaded_data_preprocessing(data_config)
+
+
+class SingleFileDset(MultiChDloader):
+    def __init__(
+        self,
+        preloaded_data: NDArray,
+        data_config: Union[MultiChDatasetConfig, LCDatasetConfig],
+        fpath: str,
+        load_data_fn: Callable[..., NDArray],
+        val_fraction=None,
+        test_fraction=None,
+    ):
+        self._preloaded_data = preloaded_data
+        super().__init__(
+            data_config,
+            fpath,
+            load_data_fn=load_data_fn,
+            val_fraction=val_fraction,
+            test_fraction=test_fraction,
+        )
+
+    def rm_bkground_set_max_val_and_upperclip_data(self, max_val, datasplit_type):
+        pass
+
+    @property
+    def data_path(self):
+        return self._fpath
+
+    def load_data(
+        self,
+        data_config: Union[MultiChDatasetConfig, LCDatasetConfig],
+        datasplit_type: DataSplitType,
+        load_data_fn: Callable[..., NDArray],
+        val_fraction=None,
+        test_fraction=None,
+        allow_generation=None,
+    ):
+        self._data = self._preloaded_data
+        assert (
+            "channel_1" not in data_config
+        ), "Outdated config file. Please remove channel_1, channel_2, channel_3 from the config file."
+        assert (
+            "channel_2" not in data_config
+        ), "Outdated config file. Please remove channel_1, channel_2, channel_3 from the config file."
+        assert (
+            "channel_3" not in data_config
+        ), "Outdated config file. Please remove channel_1, channel_2, channel_3 from the config file."
+        self._loaded_data_preprocessing(data_config)
+
+
+class MultiFileDset:
+    """
+    Here, we handle dataset having multiple files. Each file can have a different spatial dimension and number of frames (Z stack).
+    """
+
+    def __init__(
+        self,
+        data_config: Union[MultiChDatasetConfig, LCDatasetConfig],
+        fpath: str,
+        load_data_fn: Callable[..., NDArray],
+        val_fraction=None,
+        test_fraction=None,
+    ):
+        self._fpath = fpath
+        data = load_data_fn(
```

Comment:
Want to point out the inconsistency here. this file (multifile_dataset.py) is included in CAREamics. however, load_data_fn, in the current code state, will return a class object, and the class is defined outside the CAREamics (multifile_raw_dloader.py). I have trust in you that you can easily find a suitable way to resolve this.